eras/alonzo/impl/cddl-files/alonzo.cddl

```diff
@@ -111,297 +341,188 @@ script_data_hash = $hash32 ; New
-  / #6.127([* a])
-  ; similarly for tag range: 6.1280 .. 6.1400 inclusive
-  / #6.102([uint, [* a]])
-
-redeemer = [ tag: redeemer_tag, index: uint, data: plutus_data, ex_units: ex_units ] ; New
-redeemer_tag = ; New
-    0 ; inputTag "Spend"
-  / 1 ; mintTag  "Mint"
-  / 2 ; certTag  "Cert"
-  / 3 ; wdrlTag  "Reward"
-ex_units = [mem: uint, steps: uint] ; New
-
-ex_unit_prices = ; New
-  [ mem_price: sub_coin, step_price: sub_coin ]
-
-; This is an enumeration. for now there's only one value ; New
-language = 0 ; Plutus v1
-
-costmdls = { * language => cost_model } ; New
-
-; The keys to the cost model map are not present in the serialization.
-; The values in the serialization are assumed to be ordered
-; lexicographically by their correpsonding key value.
-; See Plutus' `ParamName` for parameter ordering
-cost_model = [ 166*166 int64 ] ; New
-
-transaction_metadatum =
-    { * transaction_metadatum => transaction_metadatum }
-  / [ * transaction_metadatum ]
-  / int
-  / bytes .size (0..64)
-  / text .size (0..64)
+transaction_metadatum = {* transaction_metadatum => transaction_metadatum}
+                         / [* transaction_metadatum]
+                         / int
+                         / bytes .size (0 .. 64)
+                         / text .size (0 .. 64)
 
 transaction_metadatum_label = uint
-metadata = { * transaction_metadatum_label => transaction_metadatum }
-
-auxiliary_data =
-  metadata ; Shelley
-  / [ transaction_metadata: metadata ; Shelley-ma
-    , auxiliary_scripts: [ * native_script ]
-    ]
-  / #6.259({ ? 0 => metadata         ; Alonzo and future ; New
-      , ? 1 => [ * native_script ]
-      , ? 2 => [ * plutus_script ]
-      })
-
-
-
-vkeywitness = [ $vkey, $signature ]
-
-bootstrap_witness =
-  [ public_key : $vkey
-  , signature  : $signature
-  , chain_code : bytes .size 32
-  , attributes : bytes
-  ]
-
-native_script =
-  [ script_pubkey
-  // script_all
-  // script_any
-  // script_n_of_k
-  // invalid_before
-     ; Timelock validity intervals are half-open intervals [a, b).
-     ; This field specifies the left (included) endpoint a.
-  // invalid_hereafter
-     ; Timelock validity intervals are half-open intervals [a, b).
-     ; This field specifies the right (excluded) endpoint b.
-  ]
 
-script_pubkey = (0, addr_keyhash)
-script_all = (1, [ * native_script ])
-script_any = (2, [ * native_script ])
-script_n_of_k = (3, n: uint, [ * native_script ])
+; NEW:
+;   datum_hash: $hash32
+; 
+transaction_output = [address, amount : value, ? datum_hash : $hash32]
+
+; 
+; NEW: 
+;   3: [* plutus_script ]
+;   4: [* plutus_data ]
+;   5: redeemers
+; 
+transaction_witness_set = {? 0 : [* vkeywitness]
+                          , ? 1 : [* native_script]
+                          , ? 2 : [* bootstrap_witness]
+                          , ? 3 : [* plutus_script]
+                          , ? 4 : [* plutus_data]
+                          , ? 5 : redeemers}
+
+unit_interval = #6.30([1, 2])
```

Comment:
I think this should have the following comment to avoid confusion:
```
  ; real unit_interval is: #6.30([uint, uint])
  ; but this produces numbers outside the unit interval
  ; and can also produce a zero in the denominator
```

Reply:
I guess it's related to https://github.com/input-output-hk/cuddle/issues/30

Reply:
Yes, nice suggestion! :raised_hands: